packages/nitro-protocol/contracts/ninja-nitro/SingleChannelAdjudicator.sol

```diff
@@ -308,154 +313,115 @@ contract SingleChannelAdjudicator is
+        );
+        Outcome.AssetOutcome memory gAssetOutcome = abi.decode(
+            guarantorOutcome[0].assetOutcomeBytes, // The target channel contract has already checked that all entires have the same target
+            (Outcome.AssetOutcome)
+        );
+        Outcome.Guarantee memory guarantee = abi.decode(
+            gAssetOutcome.allocationOrGuaranteeBytes,
+            (Outcome.Guarantee)
+        );
+        address targetChannelAddress = AdjudicatorFactory(adjudicatorFactoryAddress)
+            .getChannelAddress(guarantee.targetChannelId);
+        require(msg.sender == targetChannelAddress, 'only the target channel is auth');
+        bytes32 outcomeHash = keccak256(cDL.outcomeBytes);
+        _requireMatchingStorage(
+            ChannelData(
+                cDL.turnNumRecord,
+                cDL.finalizesAt,
+                cDL.stateHash,
+                cDL.challengerAddress,
+                outcomeHash
+            ),
+            guarantorChannelId
+        );
+        _transferAllAssets(payouts);
     }
 
-    function _computeNewAllocationWithGuarantee(
-        uint256 initialHoldings,
-        Outcome.AllocationItem[] memory allocation,
-        uint256[] memory indices,
-        Outcome.Guarantee memory guarantee // TODO this could just accept guarantee.destinations ?
-    )
-        public
-        pure
-        returns (
-            Outcome.AllocationItem[] memory newAllocation,
-            bool safeToDelete,
-            uint256[] memory payouts,
-            uint256 totalPayouts
-        )
-    {
-        // `indices == []` means "pay out to all"
-        // Note: by initializing payouts to be an array of fixed length, its entries are initialized to be `0`
-        payouts = new uint256[](indices.length > 0 ? indices.length : allocation.length);
-        totalPayouts = 0;
-        newAllocation = new Outcome.AllocationItem[](allocation.length);
-        safeToDelete = true; // switched to false if there is an item remaining with amount > 0
-        uint256 surplus = initialHoldings; // tracks funds available during calculation
-        uint256 k = 0; // indexes the `indices` array
-
-        // copy allocation
-        for (uint256 i = 0; i < allocation.length; i++) {
-            newAllocation[i].destination = allocation[i].destination;
-            newAllocation[i].amount = allocation[i].amount;
+    /**
+     * @notice Transfers as many funds escrowed against `guarantorChannelId` as can be afforded for a specific destination in the beneficiaries of the __target__ of that channel. Checks against the storage in this contract.
+     */
+    function claim(
+        bytes32 guarantorChannelId,
+        bytes32 targetChannelId,
+        ChannelDataAlt calldata guaranteeCDL,
+        ChannelDataAlt calldata targetCDL,
+        uint256[][] memory indices
+    ) external {
+        // CHECKS
+        for (uint256 i = 0; i < indices.length; i++) {
+            _requireIncreasingIndices(indices[i]);
```

Comment:
I believe it is possible to add redundant namespaces to functions in base contracts (such as ForceMove). So this could be
```solidity
ForceMove._requireIncreasingIndices(indices[i]);
```

Reply:
I might suggest that in the feature branch as preparation for a more thorough review in the future